modules/payloads/singles/linux/x64/shell_bind_ipv6_tcp.rb

```diff
@@ -0,0 +1,139 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/handler/bind_tcp'
+require 'msf/base/sessions/command_shell'
+require 'msf/base/sessions/command_shell_options'
+
+module MetasploitModule
+
+  CachedSize = 94
+
+  include Msf::Payload::Single
+  include Msf::Payload::Linux
+  include Msf::Sessions::CommandShellOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Linux x64 Command Shell, Bind TCP Inline (IPv6)',
+      'Description'   => 'Listen for an IPv6 connection and spawn a command shell',
+      'Author'        => 'epi <epibar052[at]gmail.com>',
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'linux',
+      'Arch'          => ARCH_X64,
+      'Handler'       => Msf::Handler::BindTcp,
+      'Session'       => Msf::Sessions::CommandShellUnix,
+      ))
+
+    def generate_stage
+      # tcp port conversion; shamelessly stolen from linux/x86/shell_reverse_tcp_ipv6.rb
+      port_order = ([1,0]) # byte ordering
+      tcp_port = [datastore['LPORT'].to_i].pack('n*').unpack('H*').to_s.scan(/../) # converts user input into integer and unpacked into a string array
+      tcp_port.pop     # removes the first useless / from  the array
+      tcp_port.shift   # removes the last useless  / from  the array
+      tcp_port = (port_order.map{|x| tcp_port[x]}).join('') # reorder the array and convert it to a string.
+      binding.pry
```

Comment:
forgot something?

Reply:
It appears so.  I removed it from [this branch/pull request](https://github.com/rapid7/metasploit-framework/pull/11047).  It looks like I need to go to git school...  I clearly jacked something up.  I wanted the two pull requests to be separate branches and separate work.  However, it didn't work out that way.  I'm sure it was user-error on my part.  Thanks for checking it over.  

I ended up getting both bind payloads in both branches exactly the same.  Sorry again for the confusion.

![not-a-smart-man](https://user-images.githubusercontent.com/43392618/50033789-9e089880-ffbf-11e8-99cc-a8436b1eadca.png)

Reply:
Along the same lines, what should I do about the [other PR](https://github.com/rapid7/metasploit-framework/pull/11047), since this one has both payloads anyway?

Reply:
We could close the other, or fix this one.  Closing the other would likely be easier, but it is up to you.